src/trusted/service_runtime/nacl_app_thread.c

```diff
@@ -146,12 +145,13 @@ struct NaClApp *NaClChildNapCtor(struct NaClApp *nap) {
   
   /* duplicate file descriptor table starting at child_fd = 3 (0-2 setup previously)*/
   NaClXMutexLock(&nap_parent->mu);
-  for (int parent_fd = nap_child->fd; parent_fd <= nap_parent->fd; parent_fd++) {
+
+  for (int fd = 0; fd <= FILE_DESC_MAX; fd++) {
 
     /* Retrive the host fd we had stored in the Cage Table for the parent */
-    int parent_host_fd = fd_cage_table[nap_parent->cage_id][parent_fd];
-    if (parent_host_fd == 0) {
-      fd_cage_table[nap_child->cage_id][nap_child->fd++] = 0;
+    int parent_host_fd = fd_cage_table[nap_parent->cage_id][fd];
+    if (parent_host_fd == NACL_BAD_FD) {
+      fd_cage_table[nap_child->cage_id][fd] = NACL_BAD_FD;
       continue;
     }
```

Comment:
This has to do with the commented out function Justin found above. I removed the separate initial descriptor setup function and replaced it with us copying over all descriptors. Because of dup functions we want new cages to inherit the previous cage's descriptors and not set up a brand new 0-2.

Reply:
Other than that, I got rid of the confusing parent and child counters and replaced it with just one fd counter that goes over all possible fds. I know this is O(n) and isn't great, but with the better fd allocation system this is the easiest way right now to make sure we copy all descriptors.

I should probably make an issue about both the allocation and copying that there are some O(n) functions that could be made more efficient when we're looking to increase speed. OTOH having an n=1024 of something that's not called very often isn't horrible.